docs/uml/protocol/em/ControlledEm.puml

```diff
@@ -0,0 +1,90 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+' set fixed order
+participant Scheduler
+participant EmAgent1
+participant EmAgent2
+participant PvAgent
+participant StorageAgent
+participant WeatherService
+
+autonumber
+
+==Init==
+
+PvAgent -> WeatherService: RegisterForWeatherMessage
+activate PvAgent
+activate WeatherService
+PvAgent -> EmAgent2: RegisterParticipant
+deactivate PvAgent
+
+WeatherService -> PvAgent: RegistrationSuccessfulMessage
+deactivate WeatherService
+activate PvAgent
+PvAgent -> EmAgent2: ScheduleFlexRequest
+deactivate PvAgent
+activate EmAgent2
+
+EmAgent2 -> EmAgent1: ScheduleFlexRequest
+deactivate EmAgent2
+activate EmAgent1
+EmAgent1 -> Scheduler: ScheduleActivation
+deactivate EmAgent1
+
+StorageAgent -> EmAgent2: RegisterParticipant
```

Comment:
Should an activation of the storage agent added here as well?

Reply:
The `StorageAgent` is scheduling itself via `ScheduleFlexRequest`. The `EmAgent2` does not need to schedule itself with `EmAgent1` again though if the second flex request (the one by StorageAgent) is scheduled for the same tick as the first one. This does not really come through with these diagrams, I didn't want to add parameters to the messages though to make the diagrams less clunky.

Reply:
Maybe I should add the ticks though to make the example clearer...